File: src/lidar_pkg/src/lidar_node.cpp
#include <ros/ros.h>
#include <sensor_msgs/LaserScan.h>
#include<geometry_msgs/Twist.h>

ros::Publisher pub;
int count = 0;
void lidar_info(const sensor_msgs::LaserScan::ConstPtr& msg)
{
    float dist = msg->ranges[180];
    ROS_INFO("前方的距离为: [%f]", dist);

    if(count>0)
        {count--;
        return ;}
    else{

    geometry_msgs::Twist twist;
    if(dist < 1.5)
    {
        count = 20;
        twist.angular.z = 0.5;
    }
    else
        twist.linear.x = 0.1;
    
    pub.publish(twist);
    }

}
int main(int argc, char  *argv[])
{

    setlocale(LC_ALL, "");
    ros::init(argc, argv,"lidar_node");

    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe("/scan", 10, lidar_info);
    pub = nh.advertise<geometry_msgs::Twist>("/cmd_vel", 10);

    ros::spin();
    return 0;
}

File: src/lidar_pkg/scripts/lidar_node.py
#!/usr/bin/env python3
import rospy
from sensor_msgs.msg import LaserScan
from geometry_msgs.msg import Twist



def laser_info(msg):
    global pub , cnt
    rospy.loginfo("前方距离为:{}".format(msg.ranges[180]))
    if cnt > 0:
        cnt -= 1
        return
    twist = Twist()
    
    if msg.ranges[180] < 1.5:
        twist.angular.z = 1.0
        cnt = 50
    else:
        twist.linear.x = 2

    pub.publish(twist)

if __name__ == '__main__':
    rospy.init_node('laser_scan')
    cnt = 50
    sub = rospy.Subscriber('/scan', LaserScan, laser_info)
    pub = rospy.Publisher('/cmd_vel', Twist, queue_size=10)
    
    rospy.spin()
    